src/ORM/Behavior/TreeBehavior.php

```diff
@@ -941,14 +943,20 @@ protected function _scope(Query $query): Query
     {
         $scope = $this->getConfig('scope');
 
+        if ($scope === null) {
+            return $query;
+        }
         if (is_array($scope)) {
             return $query->where($scope);
         }
-        if (is_callable($scope)) {
+        if ($scope instanceof Closure) {
             return $scope($query);
         }
 
-        return $query;
+        throw new UnexpectedValueException(sprintf(
+            'Scope must be an `array` or `Closure`. Got `%s` instead.',
+            get_debug_type($scope)
```

Comment:
Would like to have tests for exceptions if possible. It says closures aren't tested either?